proposals/pubsub-mq-interfaces-proposal.md

```diff
@@ -0,0 +1,144 @@
+# `pubsub` + `message-queue` Interface Proposal
+**Made 2022-11-24 by [danbugs](https://github.com/danbugs)**
+
+This document proposes a new interface for the `pubsub`, and `message-queue` capabilities together with highlighting the limitations of the current interfaces and proposing a solution.
+
+##  1. <a name='TableofContents'></a>Table of Contents
+
+<!-- vscode-markdown-toc -->
+* 1. [Table of Contents](#TableofContents)
+* 2. [Background](#Background)
+	* 2.1. [`message-queue` Background](#message-queueBackground)
+	* 2.2. [`pubsub` Background](#pubsubBackground)
+	* 2.3. [`pubsub`/`message-queue` Comparison](#pubsubmessage-queueComparison)
+* 3. [`pubsub`/`http` Interoptability](#pubsubhttpInteroptability)
+* 4. [Problem Statement](#ProblemStatement)
+* 5. [Proposal](#Proposal)
+	* 5.1. [`pubsub` WIT](#pubsubWIT)
+	* 5.2. [`slight` Runtime](#slightRuntime)
+* 6. [Alternatives](#Alternatives)
+	* 6.1. [Alternative 1](#Alternative1)
+* 7. [Additional Details](#AdditionalDetails)
+
+<!-- vscode-markdown-toc-config
+	numbering=true
+	autoSave=true
+	/vscode-markdown-toc-config -->
+<!-- /vscode-markdown-toc -->
+
+##  2. <a name='Background'></a>Background
+
+###  2.1. <a name='message-queueBackground'></a>`message-queue` Background
+
+`message-queue`s are a messaging mechanism, where message transmitters send messages to a queue construct that can be read by receivers.
```

Comment:
should mention that messages stored in a message queue can only be consumed once by one consumer.

Reply:
done ✅